extmod/asyncio/task.py

```diff
@@ -175,3 +175,76 @@ def cancel(self):
             core._task_queue.push(self)
         self.data = core.CancelledError
         return True
+
+    def add_done_callback(self, callback):
+        if not self.state:
+            callback(self, self.data)
```

Comment:
This signature of the callback does not match CPython.  In CPython the callback takes only one argument.

Reply:
That's right - I'm hoping to follow up to bring this in line with CPython in cases where only one argument is accepted in the function in a follow up PR.

However, changing the callback signature here would mean it's different from where it gets called elsewhere.

Reply:
I could add a comment to clarify that if it helps?